Assets/_Laz/Scripts/Environment/ElectricGate.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Laz
+{
+    [RequireComponent(typeof(BoxCollider))]
+    public class ElectricGate : MonoBehaviour
+    {
+
+        [SerializeField, Range(0f, 50f)] float _gateWidth = 10f;
+        [SerializeField] Transform _leftPost, _rightPost, _gate = default;
+        
+
+        private BoxCollider _gateCollider = null;
+
+        private void Awake()
+        {
+            _gateCollider = GetComponent<BoxCollider>();
+            ScaleGate();
+        }
+
+        private void OnCollisionEnter(Collision other)
+        {
+            if (other.collider.CompareTag(Tags.LazPlayer))
+            {
+                LazCoordinatorBehaviour lazCoordinator = other.gameObject.GetComponent<LazCoordinatorBehaviour>();
+
+                if (lazCoordinator != null)
+                {
+                    lazCoordinator.KillLaz();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Resizes gate according to gate width, and repositions posts
+        /// </summary>
+        private void ScaleGate()
+        {
+            if(_gate != null)
```

Comment:
So for each "component" - gate, gateCollider, rightPost. 

Is it possible for them to be empty? Since if a `gateCollider` is missing, or a `rightPost` is missing, are we okay with thatt? or should we instantly let the designers know "Hey we'rem issing a collider/post/component here, fix it, or we won't run"

Reply:
Ah okay, is there where I should use the PanicHelper you were talking about on slack?

Reply:
That's up to the developers and designers. Like should it be possible for any of these things to be missing? If it's not possible, then you can throw a panic

Reply:
Thinking about it further, since these are only getting used in edit mode anyway it makes more sense just to log a warning anyway

Reply:
alright if that's what you think . Cuz not all developers/designers would read the warning logs. and there's always a chance that they'll ignore the warning in a sea of other warnings, and it get's shipped to prod.

Reply:
Yeah I get that, but since its only getting called in OnValidate I don't think throwing a panic would do anything, and missing these and ignoring the warning is pretty low risk anyways. I think later tickets will need to manipulate the collider more so we can increase the severity of a null check there later on